backend/server.js

```diff
@@ -195,30 +247,46 @@ sbp('sbp/selectors/register', {
       // This is done in a queue to handle several simultaneous requests
       // reading and writing to the same key
       await appendToIndexFactory(resourcesKey)(resourceID)
+      sbp('chelonia.persistentActions/enqueue', ['backend/server/saveIndirectResourcesIndex', resourceID])
     })
   },
+  'backend/server/saveIndirectResourcesIndex': async function (resourceID: string) {
+    const ownerID = await sbp('chelonia.db/get', `_private_owner_${resourceID}`)
+    let indirectOwnerID = ownerID
+    while ((indirectOwnerID = await sbp('chelonia.db/get', `_private_owner_${indirectOwnerID}`))) {
+      await appendToIndexFactory(`_private_indirectResources_${indirectOwnerID}`)(resourceID)
+    }
+  },
+  'backend/server/removeFromIndirectResourcesIndex': async function (resourceID: string) {
+    const ownerID = await sbp('chelonia.db/get', `_private_owner_${resourceID}`)
+    const resources = await sbp('chelonia.db/get', `_private_resources_${resourceID}`)
+    const indirectResources = resources ? await sbp('chelonia.db/get', `_private_indirectResources_${resourceID}`) : undefined
+    const allSubresources = [
+      resourceID,
+      ...(resources ? resources.split('\x00') : []),
+      ...(indirectResources ? indirectResources.split('\x00') : [])
+    ]
+    let indirectOwnerID = ownerID
+    while ((indirectOwnerID = await sbp('chelonia.db/get', `_private_owner_${indirectOwnerID}`))) {
+      await removeFromIndexFactory(`_private_indirectResources_${indirectOwnerID}`)(allSubresources)
+    }
```

Comment:
Is this right? If the owner of the owner of the resource doesn't exist we do nothing? Or should this be a `do-while`?

Reply:
If the owner of the owner doesn't exist, there are no indirect resources.

Reply:
I think what you wrote there would make for a perfect comment